Feature/small improvements

eliaslecomte opened the pull request:
Small improvements like:
- Rename BeacinInterface to BeaconInterface.
- Use isEmpty().
- Remove unused fields.
- Reorder the modifiers to comply with the Java Language Specification.
- Correct order of variables.
- Add unit test for ConversionUtils.bytesToUuid(byteArray).
- Use Butterknife.

eliaslecomte committed:
Rename BeacinInterface to BeaconInterface.

eliaslecomte committed:
Use isEmpty().

eliaslecomte committed:
Remove unused fields.

eliaslecomte committed:
Reorder the modifiers to comply with the Java Language Specification.

eliaslecomte committed:
Correct order of variables.

eliaslecomte committed:
Add unit test for ConversionUtils.bytesToUuid(byteArray).

eliaslecomte committed:
Use Butterknife. Add edit texts to the example app so any beacon can be monitored.

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java:
```diff
@@ -3,107 +3,91 @@
+    @OnClick(R.id.button_start)
+    public void onButtonStartClicked()
+    {
+        if (isValid())
```
this?

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java:
```diff
@@ -136,7 +120,46 @@ private void beaconsEnabled(final boolean isEnabled)
+        boolean isValid = true;
+
+        final String uuid = String.valueOf(this.editTextUuid.getText());
+        if (!Pattern.matches("[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}", uuid))
```
misschien deze ergens in een util class steken?

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java:
```diff
@@ -136,7 +120,46 @@ private void beaconsEnabled(final boolean isEnabled)
+    {
+        boolean isValid = true;
+
+        final String uuid = String.valueOf(this.editTextUuid.getText());
```
kan je hier niet toString() gebruiken?

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java:
```diff
@@ -136,7 +120,46 @@ private void beaconsEnabled(final boolean isEnabled)
+            this.editTextUuid.setError(null);
+        }
+
+        final String major = String.valueOf(this.editTextMajor.getText());
```
kan je hier niet toString() gebruiken?

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java:
```diff
@@ -136,7 +120,46 @@ private void beaconsEnabled(final boolean isEnabled)
+            this.editTextMajor.setError(null);
+        }
+
+        final String minor = String.valueOf(this.editTextMinor.getText());
```
idem

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/utils/InputFilterMinMax.java:
```diff
@@ -0,0 +1,44 @@
+
+public class InputFilterMinMax implements InputFilter
+{
+    private int min, max;
```
als ik me niet vergis, staat er ergens beschreven in onze coding guidelines dat we member variablen altijd apart definiëren

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/utils/InputFilterMinMax.java:
```diff
@@ -0,0 +1,44 @@
+    {
+        try
+        {
+            int input = Integer.parseInt(dest.toString() + source.toString());
```
final :-D

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/utils/InputFilterMinMax.java:
```diff
@@ -0,0 +1,44 @@
+        return "";
+    }
+
+    private boolean isInRange(int a, int b, int c)
```
kan static gemaakt worden

Cliffus commented on app/src/main/java/mobi/inthepocket/android/beacons/app/utils/InputFilterMinMax.java:
```diff
@@ -0,0 +1,44 @@
+            int input = Integer.parseInt(dest.toString() + source.toString());
+            if (isInRange(min, max, input))
+                return null;
+        } catch (NumberFormatException nfe)
```
new line

eliaslecomte committed:
PR feedback:
- usage of final, this, new lines
- static methods
- use toString() on edit texts
